app/controllers/notifications_controller.rb

```diff
@@ -1,26 +1,32 @@
+# frozen_string_literal: true
 class NotificationsController < ApplicationController
   def index
     if params[:archive].present?
       scope = Notification.archived
     else
       scope = Notification.inbox
     end
-    @types = scope.distinct.group(:subject_type).count
-    @statuses = scope.distinct.group(:unread).count
-    @reasons = scope.distinct.group(:reason).count
+
+    @types               = scope.distinct.group(:subject_type).count
+    @statuses            = scope.distinct.group(:unread).count
+    @reasons             = scope.distinct.group(:reason).count
     @unread_repositories = scope.distinct.group(:repository_full_name).count
-    @starred = scope.starred.count
-    scope = scope.repo(params[:repo]) if params[:repo].present?
+    @starred             = scope.starred.count
+
+    scope = scope.repo(params[:repo])     if params[:repo].present?
     scope = scope.reason(params[:reason]) if params[:reason].present?
-    scope = scope.type(params[:type]) if params[:type].present?
+    scope = scope.type(params[:type])     if params[:type].present?
     scope = scope.status(params[:status]) if params[:status].present?
-    scope = scope.starred if params[:starred].present?
+    scope = scope.starred                 if params[:starred].present?
+
+    scope = scope.owned_by(current_user)
```

Comment:
This should go above the count assignments so that it only shows counts for the current user